pkgs/development/libraries/modern-cpp-kafka/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, cmake, boost, gtest, rapidjson, rdkafka }:
+
+stdenv.mkDerivation rec {
+  pname = "modern-cpp-kafka";
+  version = "2023.03.07";
+
+  src = fetchFromGitHub {
+    owner = "morganstanley";
+    repo = "modern-cpp-kafka";
+    rev = "v${version}";
+    sha256 = "sha256-7hkwM1YbveQpDRqwMZ3MXM88LTwlAT7uB8NL0t409To=";
+  };
+
+  patches = [
+    # The lib expects headers and lib is in common dir, but nix splits then into out and dev derivations
+    ./gtest_envs.patch
+  ];
+
+  # CMakeLists is looking only to /usr/lib and /opt/homebrew otherwise needs explicit paths to the libs and headers :/
```

Comment:
I think there are specific attrs for cmake flags?

Reply:
There should be, but I wasn't able to make them work. But their script supports to explicitly set envs with the paths